diff --git a/postgres-appliance/Dockerfile b/postgres-appliance/Dockerfile
index 6608483..2ef800e 100644
--- a/postgres-appliance/Dockerfile
+++ b/postgres-appliance/Dockerfile
@@ -58,7 +58,8 @@ ENV POSTGIS_VERSION=3.6 \
     PG_MON_COMMIT=88ac7b58348aa061c814982defc170644f368f39 \
     PLPROFILER=REL4_2_5 \
     PG_PROFILE=4.11 \
-    PAM_OAUTH2=v1.0.1
+    PAM_OAUTH2=v1.0.1 \
+    VCHORD_VERSION=1.1.1
 
 WORKDIR /builddeps
 RUN bash base.sh
diff --git a/postgres-appliance/build_scripts/base.sh b/postgres-appliance/build_scripts/base.sh
index bc4633d..43bc007 100644
--- a/postgres-appliance/build_scripts/base.sh
+++ b/postgres-appliance/build_scripts/base.sh
@@ -166,6 +166,14 @@ for version in $DEB_PG_SUPPORTED_VERSIONS; do
     fi
 
     # Install 3rd party stuff
+    case "$(uname -m)" in
+        aarch64) VCHORD_ARCH="arm64" ;;
+        x86_64) VCHORD_ARCH="amd64" ;;
+    esac
+    VCHORD_DEB="postgresql-${version}-vchord_${VCHORD_VERSION}-1_${VCHORD_ARCH}.deb"
+    curl -sfL "https://github.com/supervc-stack/VectorChord/releases/download/${VCHORD_VERSION}/${VCHORD_DEB}" -o "$VCHORD_DEB"
+    apt-get install -y "./$VCHORD_DEB"
+    rm -v "./$VCHORD_DEB"
 
     if [ "${TIMESCALEDB_APACHE_ONLY}" != "true" ] && [ "${TIMESCALEDB_TOOLKIT}" = "true" ]; then
         apt-get update
